codegen-server/typescript/src/main/kotlin/software/amazon/smithy/rust/codegen/server/typescript/smithy/generators/TsApplicationGenerator.kt

```diff
@@ -0,0 +1,303 @@
+                rust("    $operationName: $operationName.clone(),")
+            }
+            rust("};")
+            writer.rust("Ok(Self{ handlers })")
+        }
+    }
+
+    private fun renderAppStart(writer: RustWriter) {
+        Attribute("napi").render(writer)
+        writer.rustBlockTemplate(
+            """pub fn start(&self, socket: &TsSocket) -> #{napi}::Result<()>""",
+            *codegenScope,
+        ) {
+            rustTemplate(
+                """
+                // #{SmithyTs}::setup_tracing();
+                let plugins = #{SmithyServer}::plugin::PluginPipeline::new();
+                let builder = crate::service::PokemonService::builder_with_plugins(plugins);
+                """,
+                *codegenScope,
+            )
+            operations.map { operation ->
+                val operationName = symbolProvider.toSymbol(operation).name.toSnakeCase()
+                rust("let builder = builder.$operationName(crate::ts_operation_adaptor::$operationName);")
+            }
+            rustTemplate(
+                """
+                let app = builder.build().expect("failed to build instance of PokemonService")
+                    .layer(&#{SmithyServer}::AddExtensionLayer::new(self.handlers.clone()));
+                let service = #{tower}::util::BoxCloneService::new(app);
+                // #{SmithyTs}::server::start_hyper_worker(socket, service).expect("failed to start the hyper server");
+                start_hyper_worker(socket, service).expect("failed to start the hyper server");
+                Ok(())
+                """,
+                *codegenScope,
+            )
+        }
+    }
+
+    private fun renderSocket(writer: RustWriter) {
+        Attribute("napi").render(writer)
+        Attribute(derive(RuntimeType.Debug)).render(writer)
+        writer.rustTemplate(
+            """
+            pub struct TsSocket(#{socket2}::Socket);
+            """,
+            *codegenScope,
+        )
+
+        Attribute("napi").render(writer)
+        writer.rustBlock("impl TsSocket") {
+            writer.rust(
+                """
+                /// Create a new UNIX `Socket` from an address, port and backlog.
+                /// If not specified, the backlog defaults to 1024 connections.
+                """.trimIndent(),
+            )
+            Attribute("napi(constructor)").render(writer)
+            writer.rustBlockTemplate(
+                """pub fn new(address: String, port: i32, backlog: Option<i32>) -> #{napi}::Result<Self>""".trimIndent(),
+                *codegenScope,
+            ) {
+                writer.rustTemplate(
+                    // XXX How do I make the new_socket import here?
+                    // In the previous file, it was use aws_smithy_http_server::socket::new_socket;
+                    """
+                    Ok(Self(
+                        aws_smithy_http_server::socket::new_socket(address, port, backlog)
```

Comment:
That should be fine, you just need to use reference I believe
 ```suggestion
                        #{SmithyServer}::socket::new_socket(address, port, backlog)
```